### PR TITLE
recursively create zip file

### PR DESCRIPTION
**Pull request by MartinNowak**
- fix Issue 15222 - std.experimental omitted from phobos zip file
- avoids outdated manifest lists

### CODE REVIEW

**Comment by MartinNowak** [comment]
I didn't test the windows rule, but according to http://info-zip.org/pub/infozip/mans/zip.html the command line should be the same as on linux.

**Comment by DmitryOlshansky** [comment]
This is grand! Maybe delete MANIFEST or is it too early for that?

**Comment by CyberShadow** [comment]
> This is grand! Maybe delete MANIFEST or is it too early for that?

Isn't `MANIFEST` in Druntime?

**Comment by DmitryOlshansky** [comment]
> Isn't MANIFEST in Druntime?

Yeah, indeed. The large simplification of makefiles is still well worth it.

**Comment by MartinNowak** [comment]
We can easily extend this to druntime and get rid of some manifests.

**Comment by MartinNowak** [comment]
> Who owns @dlang-bot?

I created that user for posting cross references (see https://code.dawg.eu/writing-a-vibed-app.html), and we might integrate it with some of @CyberShadow's systems.

File: posix.mak [hunk 1]
```diff
@@ -383,7 +383,8 @@ clean :
 	rm -rf $(ROOT_OF_THEM_ALL) $(ZIPFILE) $(DOC_OUTPUT_DIR)
 
 zip :
-	zip $(ZIPFILE) $(MAKEFILE) $(ALL_D_FILES) $(ALL_C_FILES) index.d win32.mak win64.mak osmodel.mak
+	-rm -f $(ZIPFILE)
+	zip -r $(ZIPFILE) . -x .git\* -x generated\*
```
Comment (andralex) [inline posix.mak]: At least on Posix I much prefer the explicit list. Using wildcards is more error prone.

Comment (MartinNowak) [inline posix.mak]: Walter is the only one using the zip targets, so naturally they are not maintained.
Requiring everyone to maintain a list they are not using and regularly complaining that they don't doesn't make sense.

Comment (MartinNowak) [inline posix.mak]: And from what we know he uses it to sync dev work between different PCs (b/c for some weird reason installing git is a hassle), so it's actually beneficial to copy all files.
An even better idea would be to teach Walter a different workflow, e.g. `git daemon` or `rsync`, and getting rid of the zip targets alltogether.

**Comment by quickfur** [comment]
LGTM.  I agree using wildcards is more hassle-free in the long run.

File: posix.mak (hunk 1 above)
Comment (MartinNowak) [inline posix.mak]: Ping, there is no reason to maintain that list only for the zip target.

**Comment by MartinNowak** [comment]
I've updated this and also removed `ALL_C_FILES` from posix.mak, we don't need to check zlib's source code for whitespace issues.

File: posix.mak (hunk 1 above)
Comment (andralex) [inline posix.mak]: ok I guess

**Pull request merged**